# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: browser/base/content/browser-places.js
# Commit: 0a3da2090e48
# Full Hash: 0a3da2090e4859473f52b585bcb39baae2876b28
# Author: Jared Wein <jwein@mozilla.com>
# Date: 2021-02-26 03:55:47
# Regressor Bug: 1665390
# File Overlap Count: 1
# Description:
#   Bug 1665390 - Style bookmarks toolbar and tabs toolbar tooltips. r=NeilDeakin,ntim
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D104257
# ==============================================================================

diff -r 176b788753e8 -r 0a3da2090e48 browser/base/content/browser-places.js
--- a/browser/base/content/browser-places.js	Thu Feb 25 20:47:15 2021 +0000
+++ b/browser/base/content/browser-places.js	Thu Feb 25 20:47:54 2021 +0000
@@ -969,13 +969,14 @@
       return false;
     }
 
-    var tooltipTitle = aDocument.getElementById("bhtTitleText");
+    let tooltipTitle = aEvent.target.querySelector(".places-tooltip-title");
     tooltipTitle.hidden = !title || title == url;
     if (!tooltipTitle.hidden) {
-      tooltipTitle.textContent = title;
+      // Use `value` instead of `textContent` so cropping will apply
+      tooltipTitle.value = title;
     }
 
-    var tooltipUrl = aDocument.getElementById("bhtUrlText");
+    let tooltipUrl = aEvent.target.querySelector(".places-tooltip-uri");
     tooltipUrl.hidden = !url;
     if (!tooltipUrl.hidden) {
       tooltipUrl.value = url;